apps/vscode/src/providers/preview/preview-util.ts

```diff
@@ -59,6 +59,30 @@ export function isQuartoShinyKnitrDoc(
   
 }
 
+export async function isRPackage() {
```

Comment:
This code is similar to what we are using elsewhere to detect if we are in an R package; it is working well for us so far. Consider it analogous `isQuartoShinyDoc()` in this same file, although it applies to the _workspace_ rather than a file. We believe it is reasonable to make assumptions about folks using workspaces for R packages.

Reply:
The code looks good, but it doesn't look like auto-formatting is working in this repo, and this uses tabs 😱.

Reply:
I'm very happy to help get some minimal `settings.json` set up in this repo, or of course for you to do it and then adjust. If I and/or other people are going to be contributing more, it probably makes sense to do that pretty soon.

Reply:
I think it makes sense to use whatever `settings.json` is common across the rest of the company. `quarto-cli` is a Deno project so it uses `deno.jsonc` instead. I'd be happy for `quarto` to take that `settings.json` in a separate PR.